core/src/main/java/nl/nn/adapterframework/webcontrol/api/SendJmsMessage.java

```diff
@@ -212,4 +192,51 @@ private void processMessage(JmsSender qms, String messageId, String message) thr
 			throw new ApiException("Error occured on closing connection", e);
 		}
 	}
+	
+	
+	private Optional<String> getAttributeValue(String name, List<Attachment> attachmentList) {
+		return this.getFile(name, attachmentList).map(v -> {
+			return Optional.ofNullable(v.toString());
+		}).orElse(Optional.empty());
+	}
+	
+	private Optional<InputStream> getFile(String name, List<Attachment> attachmentList) {
+		return attachmentList.stream().filter(att -> name.equalsIgnoreCase(att.getDataHandler().getDataSource().getName())).findAny().map(m -> {
```

Comment:
This written out in normal style java:

for(Attachment attachment: attachmentList) {
  if(name == attachment.getName()) return attachment.getInputStream();
}

reads a lot nicer...

Reply:
we need to take advantage from stream, in this case, besides to iterate, i need to filter by "name". also, using optional we can avoid nullpionterexception :)